### PR TITLE
Show sound sources

### LINKED ISSUE #1270: Show sound sources
Show sound sources in a similar way to camera/sinks.
Add the beautiful cube.
Show which one is played in some way - not sure how they're named but try to show something useful.

### PR DESCRIPTION
**Pull request by chreden**
Add sound sources to the viewer.
Add a sound sources window. Window also shows the sounds in the level (sound board) and allows user to play either a specific sound source sample or a sound from the board.
Attempts to load from MAIN.SFX if present in the same folder for TR2 and TR3.
Closes #1270

### CODE REVIEW

File: trlevel/Level.cpp [hunk 1]
```diff
@@ -892,14 +900,28 @@ namespace trlevel
         }
         else
         {
-            log_file(activity, file, "Skipping uncompresed and compressed sizes - unused in Tomb5");
-            skip(file, 8);
+            log_file(activity, file, "Reading uncompresed size and skipping compressed size - unused in Tomb5");
```
Comment (lahm86) [inline trlevel/Level.cpp:903]: Typo - `uncompresed`

**Review by lahm86 — COMMENTED** [review]
TR1, TR4 and TR5 look good to me. Classic TR2/3 both unfortunately crash as follows (but are fine when opening from a folder that doesn't have MAIN.SFX).

image

image

This doesn't happen if I open remastered files, so I assume it's a slight difference in the MAIN.SFX layout. I remember looking at this and put the hack below in place, but I promptly  forgot about it as it's a rarely used utility.

https://github.com/LostArtefacts/TR-Rando/blob/master/SFXExport/Program.cs#L54

**Comment by lahm86** [comment]
Maybe a slight improvement (possibly a future task) would be to show when the sound data is not present in the level file. ID 304 isn't in the sound map in this level, but with the sample showing 0 it's sort of implied that should be Lara's footsteps. Maybe some sort of indicator would be good here.

Screenshot 2024-08-23 104045

**Comment by chreden** [comment]
> TR1, TR4 and TR5 look good to me. Classic TR2/3 both unfortunately crash as follows (but are fine when opening from a folder that doesn't have MAIN.SFX).
> 
> image
> 
> image
> 
> This doesn't happen if I open remastered files, so I assume it's a slight difference in the MAIN.SFX layout. I remember looking at this and put the hack below in place, but I promptly forgot about it as it's a rarely used utility.
> 
> https://github.com/LostArtefacts/TR-Rando/blob/master/SFXExport/Program.cs#L54

Ah thanks. Might be a bit of an issue emulating that `if (remaster)` part...

**Review by lahm86 — APPROVED** [review]

**Pull request merged**